pkg/scheduler/objects/queue_test.go

```diff
@@ -616,29 +616,83 @@ func TestSortQueue(t *testing.T) {
 	// empty parent queue
 	parent, err = createManagedQueue(root, "parent", true, nil)
 	assert.NilError(t, err, "failed to create parent queue")
-	if len(parent.sortQueues()) != 0 {
-		t.Errorf("parent queue should return sorted queues: %v", parent)
+	if len(parent.getSortedQueues()) != 0 {
+		t.Errorf("parent queue shouldn't return sorted queues: %v", parent)
 	}
 
 	// leaf queue must be nil
 	leaf, err = createManagedQueue(parent, "leaf", false, nil)
 	assert.NilError(t, err, "failed to create leaf queue")
-	if queues := leaf.sortQueues(); queues != nil {
-		t.Errorf("leaf queue should return sorted queues: %v", queues)
+	if queues := leaf.getSortedQueues(); queues != nil {
+		t.Errorf("leaf queue shouldn't return sorted queues: %v", queues)
 	}
-	if queues := parent.sortQueues(); len(queues) != 0 {
+	if queues := parent.getSortedQueues(); len(queues) != 0 {
 		t.Errorf("parent queue with leaf returned unexpectd queues: %v", queues)
 	}
+
+	// increase pending resoruce from 0 to 1
 	var res *resources.Resource
 	res, err = resources.NewResourceFromConf(map[string]string{"first": "1"})
 	assert.NilError(t, err, "failed to create basic resource")
 	leaf.incPendingResource(res)
-	if queues := parent.sortQueues(); len(queues) != 1 {
+	if queues := parent.getSortedQueues(); len(queues) != 1 {
 		t.Errorf("parent queue did not return expected queues: %v", queues)
 	}
+	if !reflect.DeepEqual(parent.sortedQueues, parent.getSortedQueues()) {
+		t.Errorf("parent queue did not use cached sorted queues: %v", parent)
+	}
```

Comment:
I know it's based on existing code, but let's use `assert.Equal()` or `assert.Assert()`, there's no need for `t.Errorf()`. It's much cleaner.